src/init.cpp

```diff
@@ -1468,8 +1468,18 @@ bool AppInit2(boost::thread_group& threadGroup, CScheduler& scheduler)
 #endif // !ENABLE_WALLET
 
 #ifdef ENABLE_MINING
+ #ifndef ENABLE_WALLET
+    auto mineToLocalWallet = GetBoolArg("-minetolocalwallet", false);
+    if (mineToLocalWallet) {
+        return InitError(_("Zcash was not built with wallet support. Set -minetolocalwallet=0 to use -mineraddress, or rebuild Zcash with wallet support."));
+    }
+    if (!mapArgs.count("-mineraddress")) {
+        return InitError(_("Zcash was not built with wallet support. Set -mineraddress, or rebuild Zcash with wallet support."));
```

Comment:
As-is, this will require a build with `!ENABLE_WALLET && ENABLE_MINING` to always have `-mineraddress` set, even if `-gen` is not set. Is this what we want? It does make some sense, because `zcash-cli setgenerate` wouldn't work properly if `-mineraddress` wasn't set, and the user likely does want to mine in this setup, but I'm not sure we should assume that.